file/file_prefetch_buffer.cc

```diff
@@ -315,17 +315,41 @@ Status FilePrefetchBuffer::PrefetchAsyncInternal(
 
     // Move data from curr_ buffer to third.
     CopyDataToBuffer(curr_, offset, length);
+    // swap the buffers.
+    curr_ = curr_ ^ 1;
+    second = curr_ ^ 1;
+
+    // All the bytes are in curr_. second buffer can go for async prefetching
```

Comment:
Not sure I understand this. This whole `if` block is for the case where the required data overlaps curr_ and second_ right? So second_ may not necessarily be empty.